doc/sphinx/proof-engine/vernacular-commands.rst

```diff
@@ -568,67 +476,50 @@ Compiled files
 
 This section describes the commands used to load compiled files (see
 Chapter :ref:`thecoqcommands` for documentation on how to compile a file). A compiled
-file is a particular case of module called *library file*.
-
-
-.. cmd:: Require @qualid
-
-   This command looks in the loadpath for a file containing module :n:`@qualid`
-   and adds the corresponding module to the environment of |Coq|. As
-   library files have dependencies in other library files, the command
-   :cmd:`Require` :n:`@qualid` recursively requires all library files the module
-   qualid depends on and adds the corresponding modules to the
-   environment of |Coq| too. |Coq| assumes that the compiled files have been
-   produced by a valid |Coq| compiler and their contents are then not
-   replayed nor rechecked.
-
-   To locate the file in the file system, :n:`@qualid` is decomposed under the
-   form :n:`dirpath.@ident` and the file :n:`@ident.vo` is searched in the physical
-   directory of the file system that is mapped in |Coq| loadpath to the
-   logical path dirpath (see Section :ref:`libraries-and-filesystem`). The mapping between
-   physical directories and logical names at the time of requiring the
-   file must be consistent with the mapping used to compile the file. If
-   several files match, one of them is picked in an unspecified fashion.
+file is a particular case of a module called a *library file*.
+
 
+.. cmd:: Require {? {| Import | Export } } {+ @qualid }
+   :name: Require; Require Import; Require Export
 
-   .. cmdv:: Require Import @qualid
-      :name: Require Import
+   Loads compiled modules into the |Coq| environment.  For each :n:`@qualid`, which has the form
+   :n:`{* @ident__module . } @ident`, the command searches for the logical name represented
+   by the :n:`ident__module`\s and loads the compiled file :n:`@ident.vo` from the associated
```

Comment:
Got it.  To be precise, in the fully-qualified module name `Coq.Init.Nat`, `Coq` is not the name of a filesystem directory.  There is a tight relationship between filesystem directories and module names but they are not the same thing.  Maybe once the loadpath section is redone we'll have a precise defined term that we can use here.

Reply:
Right. Then "module prefix" or just "prefix" is the best we can do.

Reply:
That being said, even if it is not a directory name initially, `Coq` is matched to a directory. And the rest of Coq packages, when they are installed, go to a directory named according to this prefix, within the `user-contrib` directory.

Reply:
I made it "prefix".  But the `Print LoadPath` output is puzzling.
- Shouldn't the 3 lines with `<>` give the first component of the fully-qualified name?
- The reverse-alpha ordering is strange.
- `Locate Library VST.veric` fails (not loaded, but IIUC should still report on this)
- `Locate Library firstorder` fails even though it's in the list
- Would be nice for `Locate Library Init` to say it's a directory or list the libraries and directories therein. (enhancement)
- `Locate firstorder` reports "Ltac Coq.Init.Prelude.firstorder"
- Would be nice to indicate what's loaded

```
<> C:\Coq8.10.1\lib\coq\user-contrib
VST.veric C:\Coq8.10.1\lib\coq\user-contrib\VST\veric
VST.sepcomp C:\Coq8.10.1\lib\coq\user-contrib\VST\sepcomp
VST.progs C:\Coq8.10.1\lib\coq\user-contrib\VST\progs
  : 
Coq C:\Coq8.10.1\lib\coq\plugins
Coq.firstorder C:\Coq8.10.1\lib\coq\plugins\firstorder

Locate firstorder.

```

Reply:
The lines with `<>` should mean that the first component of the fully-qualified name will be the name of a sub-directory of this directory. But I agree that the output is puzzling and that I get things that I shouldn't see in there.

`Locate firstorder` locates the tactic of that name, i.e. it is equivalent to `Locate Ltac firstorder`.

As for `Locate Library` it only locates Coq modules, not prefixes. And there is no module of name `firstorder`. But `Locate Library omega.Omega.` or `Locate Library Omega.` do work.

Reply:
> there is no module of name `firstorder`.

But this sure looks like a module name:

`Coq.firstorder C:\Coq8.10.1\lib\coq\plugins\firstorder`

Outside of this PR, I'd like to make the output of `Print Loadpath` more understandable.  And perhaps at the same time give further consideration to the statement "If several files match, one of them is picked in an unspecified fashion."

Reply:
This is a module prefix (or "dirpath"). Everything in the output of `Print LoadPath` is a module prefix rather than a module.

> Outside of this PR, I'd like to make the output of `Print Loadpath` more understandable. And perhaps at the same time give further consideration to the statement "If several files match, one of them is picked in an unspecified fashion."

This is not a commonly used command, but still useful for debugging I guess, so sure. We can discuss this in more details in a separate issue.

Reply:
See #12061.